lib/msal-browser/src/app/ClientApplication.ts

```diff
@@ -771,5 +810,36 @@ export abstract class ClientApplication {
         return validLogoutRequest;
     }
 
+    /**
+     * Broadcasts events by calling callback with broadcast message
+     * @param eventType 
+     * @param interactionType 
+     * @param payload 
+     * @param error 
+     */
+    broadcastEvent(eventType: BroadcastEvent, interactionType?: InteractionType, payload?: BroadcastPayload, error?: BroadcastError) {
+        const message: BroadcastMessage = {
+            eventType: eventType,
+            interactionType: interactionType || null,
+            payload: payload || null,
+            error: error || null,
+            timestamp: Date.now()
+        };
+
+        this.eventCallbacks.forEach((callback) => {
+            callback(message);
+        });
+    }
+
+    /**
+     * Adds event callbacks to array
+     * @param callback 
+     */
+    addEventCallback(callback: Function) {
```

Comment:
I think a type should be added for the callback that shows the parameter it will be passed. I think it would be:

`export type EventCallbackFunction = (message: BroadcastMessage) => void;`

Reply:
Done.